src/machinery/wallet.ts

```diff
@@ -1,38 +1,43 @@
 import { wallet } from 'nanocurrency-web';
 import type { NanoWallet } from './models';
-import { destroyWallet, setWallet } from './secure-storage';
+import { setWallet } from './secure-storage';
 
 export interface WalletResult {
   seed: string;
   mnemonic: string;
 }
 
-export function generateWallet(): WalletResult {
-  const w = wallet.generate();
-  return {
-    seed: w.seed,
-    mnemonic: w.mnemonic,
-  };
+export function generateWallet(): Promise<WalletResult> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
```

Comment:
What is this setTimeout for?

Reply:
To be able to delay the function, so that we can show the loading screen. There might be a more ideomatic way to do this though..